### PR TITLE
Add `make install` target

### LINKED ISSUE #120: Add `make install` target

### PR DESCRIPTION
**Pull request by kpcyrd**
Pinging @adityasaky and @jvoisin for feedback. :)

Resolves #120

### CODE REVIEW

**Review by adityasaky — APPROVED** [review]
Looks good to me! Should make install also setup the man files?

File: Makefile [hunk 1]
```diff
@@ -6,4 +6,27 @@ docs: contrib/docs/rebuilderd.1 contrib/docs/rebuildctl.1 contrib/docs/rebuilder
 contrib/docs/%: contrib/docs/%.scd
 	scdoc < $^ > $@
 
-.PHONY: all docs
+install:
```
Comment (jvoisin) [inline Makefile:9]: `install: all docs` would make it sure that things are built before trying to be installed.

File: Makefile [hunk 2]
```diff
@@ -6,4 +6,27 @@ docs: contrib/docs/rebuilderd.1 contrib/docs/rebuildctl.1 contrib/docs/rebuilder
...
+	@if [ ! -e target/release/rebuilderd ]; then \
+		echo >&2 "No binary found, run make first"; \
+		false; \
+	fi
+	install -Dm 755 -t "$(DESTDIR)/usr/bin/" \
+		target/release/rebuilderd \
+		target/release/rebuildctl \
+		target/release/rebuilderd-worker
+	install -Dm 755	worker/rebuilder-*.sh -t "$(DESTDIR)/usr/libexec/rebuilderd/" \
+	for x in rebuilderd.conf rebuilderd-sync.conf rebuilderd-worker.conf; do \
```
Comment (jvoisin) [inline Makefile:19]: Maybe name it `f` for file, instead of `x`?

File: Makefile [hunk 3]
```diff
@@ -6,4 +6,27 @@ docs: contrib/docs/rebuilderd.1 contrib/docs/rebuildctl.1 contrib/docs/rebuilder
...
+		test -e "$(DESTDIR)/etc/$$x" || install -Dm 644 "contrib/confs/$$x" -t "$(DESTDIR)/etc" ; \
```
Comment (jvoisin) [inline Makefile:20]: Shouldn't existing files sometimes be overwritten?

**Pull request merged**